Yank/sampling.py

```diff
@@ -789,7 +789,10 @@ def _initialize_netcdf(self):
 
     def _write_iteration_netcdf(self):
         super(ModifiedHamiltonianExchange, self)._write_iteration_netcdf()
-        if self.fully_interacting_state is not None:
+        if self.mpicomm:
+            # Only the root node will write data.
+            if self.mpicomm.rank != 0: return
```

Comment:
I've already added this in PR #442 with another couple of bugfixes. I wanted to implement also the detection of `exclusive` vs `shared` mode before merging, but I can do it right now and you can debug from that one so that we'll avoid having to solve conflicts.